# Gikomba Stall Analysis

ROBIN KINYUA  
HPF COHORT

This notebook analyses Wanjiku's sales and stock data from her fresh-produce stall at Gikomba Market.

In [1]:
daily_sales = [2300, 1950, 2800, 2100, 3600, 5200, 4100,
               2500, 2200, 3100, 2400, 3900, 5600, 4700]   # 14 days, KES

days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"] * 2

# qty = kg in stock, reorder = reorder level (kg), cost/price = KES per kg
stock = {
    "Tomatoes": {"qty": 18, "reorder": 25, "cost": 60, "price": 90},
    "Onions":   {"qty": 40, "reorder": 20, "cost": 70, "price": 100},
    "Sukuma":   {"qty": 9,  "reorder": 15, "cost": 30, "price": 50},
    "Potatoes": {"qty": 55, "reorder": 30, "cost": 50, "price": 75},
    "Avocado":  {"qty": 12, "reorder": 10, "cost": 80, "price": 140},
}

## Task 1: Overall Sales Performance

I will calculate the total, average, highest and lowest sales for the 14 days. I will also identify the day and week of the best and worst sales.

In [2]:
total_sales = sum(daily_sales)
average_sales = total_sales / len(daily_sales)
highest_sales = max(daily_sales)
lowest_sales = min(daily_sales)

best_position = daily_sales.index(highest_sales)
worst_position = daily_sales.index(lowest_sales)

best_day = days[best_position]
worst_day = days[worst_position]

best_week = best_position // 7 + 1
worst_week = worst_position // 7 + 1

print(f"Total sales: KES {total_sales:,}")
print(f"Average daily sales: KES {average_sales:,.2f}")
print(f"Highest sales: KES {highest_sales:,} on {best_day}, Week {best_week}")
print(f"Lowest sales: KES {lowest_sales:,} on {worst_day}, Week {worst_week}")

Total sales: KES 46,450
Average daily sales: KES 3,317.86
Highest sales: KES 5,600 on Sat, Week 2
Lowest sales: KES 1,950 on Tue, Week 1


## Task 2: Week on Week Growth

I will create a function that uses slicing to calculate the total sales for Week 1 and Week 2. I will then calculate the percentage growth from Week 1 to Week 2.

In [3]:
def week_totals(sales):
    """Return the total sales for week 1 and week 2."""
    week1 = sum(sales[:7])
    week2 = sum(sales[7:])
    return week1, week2


week1_total, week2_total = week_totals(daily_sales)

growth_pct = (week2_total - week1_total) / week1_total * 100

print(f"Week 1 total: KES {week1_total:,}")
print(f"Week 2 total: KES {week2_total:,}")
print(f"Week-on-week growth: {growth_pct:.1f}%")

Week 1 total: KES 22,050
Week 2 total: KES 24,400
Week-on-week growth: 10.7%


## Task 3: Classify Sales Days

I will create a function that classifies each day as Strong, Normal or Weak by comparing the day's sales with the overall average. I will then count how many days fall into each category.

In [4]:
def classify_day(amount, average):
    """Return Strong, Normal or Weak based on the average sales."""
    if amount > average * 1.20:
        return "Strong"
    elif amount < average * 0.80:
        return "Weak"
    else:
        return "Normal"


category_counts = {}

for sale in daily_sales:
    category = classify_day(sale, average_sales)
    category_counts[category] = category_counts.get(category, 0) + 1

print(category_counts)

{'Weak': 6, 'Normal': 4, 'Strong': 4}


## Task 4: Weekday and Weekend Performance

I will separate weekday sales from weekend sales, calculate the average for each group and compare the difference as a percentage.

In [5]:
weekday_sales = []
weekend_sales = []

for i in range(len(daily_sales)):
    if days[i] in ["Mon", "Tue", "Wed", "Thu", "Fri"]:
        weekday_sales.append(daily_sales[i])
    else:
        weekend_sales.append(daily_sales[i])

weekday_average = sum(weekday_sales) / len(weekday_sales)
weekend_average = sum(weekend_sales) / len(weekend_sales)

difference_pct = (weekend_average - weekday_average) / weekday_average * 100

print(f"Average weekday sales: KES {weekday_average:,.2f}")
print(f"Average weekend sales: KES {weekend_average:,.2f}")
print(f"Weekend sales are {difference_pct:.1f}% higher than weekday sales.")

Average weekday sales: KES 2,685.00
Average weekend sales: KES 4,900.00
Weekend sales are 82.5% higher than weekday sales.


## Task 5: Stock and Gross Margin

I will check the stock levels to find products that are at or below their reorder level. I will also calculate the gross margin percentage for each product.

In [6]:
def products_to_reorder(stock_data):
    """Return a list of products whose quantity is at or below the reorder level."""
    reorder_list = []

    for product, info in stock_data.items():
        if info["qty"] <= info["reorder"]:
            reorder_list.append(product)

    return reorder_list


reorder_products = products_to_reorder(stock)

print("Products to reorder:", reorder_products)

for product, info in stock.items():
    margin_pct = (info["price"] - info["cost"]) / info["price"] * 100
    print(f"{product}: {margin_pct:.1f}% gross margin")

Products to reorder: ['Tomatoes', 'Sukuma']
Tomatoes: 33.3% gross margin
Onions: 30.0% gross margin
Sukuma: 40.0% gross margin
Potatoes: 33.3% gross margin
Avocado: 42.9% gross margin


## Task 6: Stall Performance Report

I will combine the main results into one clear report using f-strings. All money values will be formatted in KES with thousands separators.

In [7]:
print("=" * 45)
print("STALL PERFORMANCE REPORT")
print("=" * 45)

print(f"Total sales:             KES {total_sales:,}")
print(f"Average daily sales:     KES {average_sales:,.2f}")
print(f"Highest daily sales:     KES {highest_sales:,}")
print(f"Lowest daily sales:      KES {lowest_sales:,}")
print(f"Week 1 sales:            KES {week1_total:,}")
print(f"Week 2 sales:            KES {week2_total:,}")
print(f"Week-on-week growth:     {growth_pct:.1f}%")
print(f"Weekday average:         KES {weekday_average:,.2f}")
print(f"Weekend average:         KES {weekend_average:,.2f}")
print(f"Weekend difference:      {difference_pct:.1f}%")
print(f"Products to reorder:     {', '.join(reorder_products)}")

print("=" * 45)

STALL PERFORMANCE REPORT
Total sales:             KES 46,450
Average daily sales:     KES 3,317.86
Highest daily sales:     KES 5,600
Lowest daily sales:      KES 1,950
Week 1 sales:            KES 22,050
Week 2 sales:            KES 24,400
Week-on-week growth:     10.7%
Weekday average:         KES 2,685.00
Weekend average:         KES 4,900.00
Weekend difference:      82.5%
Products to reorder:     Tomatoes, Sukuma


## Recommendations for Wanjiku

Wanjiku's stall made **KES 46,450** over the 14 days, and sales increased by **10.7%** from Week 1 to Week 2, showing positive growth. Weekend sales averaged **KES 4,900**, which was about **82.5% higher** than the weekday average of **KES 2,685**, so she should make sure enough stock is available for Saturdays and Sundays. Tomatoes and Sukuma should be restocked because their current quantities are at or below their reorder levels. Avocado has the highest gross margin at about **42.9%**, so keeping enough avocado stock could help improve profitability. She should also focus on improving weaker weekdays because **6 of the 14 days** were classified as Weak.